arangod/RestHandler/RestIndexHandler.cpp

```diff
@@ -193,13 +197,89 @@ RestStatus RestIndexHandler::getIndexes() {
       return RestStatus::DONE;
     }
 
+    std::stringstream ap;
+    ap << "Plan/Collections/" << _vocbase.name() << "/" << coll->planId()
+       << "/indexes";
+    auto& ac = _vocbase.server().getFeature<ClusterFeature>().agencyCache();
+    auto [plannedIndexes, idx] = ac.get(ap.str());
+
     TRI_ASSERT(indexes.slice().isArray());
     VPackBuilder tmp;
     tmp.openObject();
     tmp.add(StaticStrings::Error, VPackValue(false));
     tmp.add(StaticStrings::Code,
             VPackValue(static_cast<int>(ResponseCode::OK)));
-    tmp.add("indexes", indexes.slice());
+    tmp.add(VPackValue("indexes"));
+    {
+      VPackArrayBuilder guard(&tmp);
+      for (auto const& i : VPackArrayIterator(indexes.slice())) {
+        tmp.add(i);
+      }
+      for (auto const& pi : VPackArrayIterator(plannedIndexes->slice())) {
+        if (pi.hasKey("isBuilding") && pi.get("isBuilding").getBool()) {
+          VPackObjectBuilder o(&tmp);
+          for (auto const& source : VPackObjectIterator(pi)) {
+            tmp.add(source.key.copyString(), source.value);
+          }
+          std::string iid = pi.get("id").copyString();
+          double progress = 0;
+          double success = 0;
+          auto const shards = coll->shardIds();
+          auto const body = VPackBuffer<uint8_t>();
+          auto* pool =
+              coll->vocbase().server().getFeature<NetworkFeature>().pool();
+          std::vector<Future<network::Response>> futures;
+          futures.reserve(shards->size());
+          std::string const prefix = "/_api/index/";
+          network::RequestOptions reqOpts;
```

Comment:
The requests will always be executed inside the system database, which will not work if the underlying collection is executed in a different database.
The setup of the RequestOptions is missing the `database` attribute here, i.e. `reqOptions.database = _vocbase.name();`

Reply:
Done in the new place.